### PR TITLE
Store hashed content in db

### LINKED ISSUE #79: Don't store full post content on blockchain

### PR DESCRIPTION
**Pull request by vivianjeng**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- Closes #79 
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/core/contracts/UnirepSocial.sol [hunk 1]
```diff
@@ -46,15 +46,15 @@ contract UnirepSocial {
     event PostSubmitted(
         uint256 indexed _epoch,
         uint256 indexed _epochKey,
-        string _postContent,
+        bytes32 _postContent,
```
Comment (vimwitch) [inline packages/core/contracts/UnirepSocial.sol:49]: we should change these args to be something like `_contentHash` just to mark these as being hashes and not full content.

File: packages/core/src/synchronizer.ts [hunk 2]
```diff
@@ -138,11 +138,11 @@ export class UnirepSocialSynchronizer extends Synchronizer {
...
             event.data
         )
-        const _transactionHash = event.transactionHash
+        const transactionHash = event.transactionHash
         const commentId = event.transactionHash
         const postId = event.topics[2]
-        const _epoch = Number(event.topics[1])
-        const _epochKey = BigInt(event.topics[3]).toString(16)
+        const epoch = Number(event.topics[1])
+        const epochKey = BigInt(event.topics[3]).toString(16)
```
Comment (vimwitch) [inline packages/core/src/synchronizer.ts:145]: I think we should store these as base 10, or base 16 with a `0x` prefix. If we do it like this we get a string that is base 16 with no `0x` prefix. We have no way to decode these types of strings properly.

We've been moving toward storing as base10 in other pr's.

File: packages/core/src/synchronizer.ts [hunk 3]
```diff
@@ -243,9 +244,9 @@ export class UnirepSocialSynchronizer extends Synchronizer {
             'PostSubmitted',
             event.data
         )
-        const _transactionHash = event.transactionHash
-        const _epoch = Number(event.topics[1])
-        const _epochKey = BigInt(event.topics[2]).toString(16)
+        const transactionHash = event.transactionHash
+        const epoch = Number(event.topics[1])
+        const epochKey = BigInt(event.topics[2]).toString(16)
```
Comment (vimwitch) [inline packages/core/src/synchronizer.ts:249]: base 10 instead

File: packages/core/src/synchronizer.ts [hunk 4]
```diff
@@ -157,16 +157,17 @@ export class UnirepSocialSynchronizer extends Synchronizer {
                 },
                 update: {
                     status: 1,
-                    transactionHash: _transactionHash,
+                    transactionHash,
                 },
             })
         } else {
+            const hashedContent: string = decodedData._postContent ?? ''
```
Comment (vimwitch) [inline packages/core/src/synchronizer.ts:164]: We should never get an `undefined` or `null` post content hash. We should also check the hash against the actual content to make sure it matches.

**Review by vimwitch — CHANGES_REQUESTED** [review]

File: packages/backend/src/routes/PostRouter.ts [hunk 5]
```diff
@@ -105,13 +105,16 @@ async function createPost(req, res) {
...
     // Parse Inputs
-    const { publicSignals, proof } = req.body
+    const { publicSignals, proof, title, content } = req.body
     const reputationProof = new ReputationProof(
         publicSignals,
         formatProofForSnarkjsVerification(proof)
     )
     const epochKey = reputationProof.epochKey.toString()
     const minRep = Number(reputationProof.minRep)
+    const hashedContent = ethers.utils.keccak256(
```
Comment (vimwitch) [inline packages/backend/src/routes/PostRouter.ts:115]: I'm torn on this, maybe we should use poseidon so it's zk friendly?

Comment (vivianjeng) [reply packages/backend/src/routes/PostRouter.ts:115]: if we don't verify hash content in a zk proof we don't have to use poseidon
and now unirep social doesn't install the poseidon package (but there is `ethers` in the unirep social)
I prefer to use `keccak256`

**Review by vimwitch — APPROVED** [review]

**Pull request merged**